### ISSUE TITLE
classify/train.py script should accept patience parameter

### ISSUE THREAD

**Issue by sohang3112** [ultralytics/yolov5#11491]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

[This Stack Overflow answer gives a good explanation of what patience is](https://stackoverflow.com/a/59590438/12947681)

Suppose a model is being trained for 100 epochs. But after epoch 60, the model's accuracy & loss don't improve any more.
If 5 more epochs pass and the model's accuracy & loss still don't improve, then the training loop should stop running.

In this example, the best epoch is 60, and patience is 5 epochs (number of epochs for which training loop should continue running when accuracy & loss stop improving).

### Use case

This would save time during model training, as the training loop will not keep running for no reason even when model's loss is not decreasing (or even increasing!)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@sohang3112 thank you for suggesting this feature, it could definitely be a useful addition to YOLOv5. We appreciate your willingness to contribute to the project by submitting a PR. Our team will review and consider your idea for implementation in future updates.

**Comment by sohang3112** [comment]
Unlike `classify/train.py`, `train.py` *has* implemented patience using `utils.torch_utils.EarlyStopping` class. Will try to use this approach for implementing in `classify/train.py` also.  A slight modification in `EarlyStopping` class would also be needed - its logging message would have to print `classify/train.py` instead of `train.py`.

**Comment by glenn-jocher** [comment]
@sohang3112 thank you for the suggestion, we appreciate your input. It's great to hear that patience has already been implemented in the `train.py` script using the `utils.torch_utils.EarlyStopping` class. We will certainly consider your idea for implementing patience in `classify/train.py` using a similar approach and modifying the logging message to print the correct script name. Your contribution to the YOLOv5 community is valuable, and we thank you for your help to improve the project.
